internal/controller/factory/statefulset_test.go

```diff
@@ -129,8 +130,144 @@ var _ = Describe("CreateOrUpdateStatefulSet handler", func() {
 			By("Checking the extraArgs")
 			Expect(sts.Spec.Template.Spec.Containers[0].Command).To(Equal(generateEtcdCommand(etcdcluster)))
 
-			By("Deleting the statefulset")
-			Expect(k8sClient.Delete(ctx, sts)).To(Succeed())
+			By("Checking the default startup probe", func() {
+				Expect(sts.Spec.Template.Spec.Containers[0].StartupProbe).To(Equal(&v1.Probe{
+					ProbeHandler: v1.ProbeHandler{
+						HTTPGet: &v1.HTTPGetAction{
+							Path:   "/readyz?serializable=false",
+							Port:   intstr.FromInt32(2379),
+							Scheme: v1.URISchemeHTTP,
+						},
+					},
+					InitialDelaySeconds: 1,
+					TimeoutSeconds:      1,
+					PeriodSeconds:       5,
+					SuccessThreshold:    1,
+					FailureThreshold:    3,
+				}))
+			})
+
+			By("Checking the default readiness probe", func() {
+				Expect(sts.Spec.Template.Spec.Containers[0].ReadinessProbe).To(Equal(&v1.Probe{
+					ProbeHandler: v1.ProbeHandler{
+						HTTPGet: &v1.HTTPGetAction{
+							Path:   "/readyz",
+							Port:   intstr.FromInt32(2379),
+							Scheme: v1.URISchemeHTTP,
+						},
+					},
+					InitialDelaySeconds: 5,
+					TimeoutSeconds:      1,
+					PeriodSeconds:       5,
+					SuccessThreshold:    1,
+					FailureThreshold:    3,
+				}))
+			})
+
+			By("Checking the default liveness probe", func() {
+				Expect(sts.Spec.Template.Spec.Containers[0].LivenessProbe).To(Equal(&v1.Probe{
+					ProbeHandler: v1.ProbeHandler{
+						HTTPGet: &v1.HTTPGetAction{
+							Path:   "/livez",
+							Port:   intstr.FromInt32(2379),
+							Scheme: v1.URISchemeHTTP,
+						},
+					},
+					InitialDelaySeconds: 5,
+					TimeoutSeconds:      1,
+					PeriodSeconds:       5,
+					SuccessThreshold:    1,
+					FailureThreshold:    3,
+				}))
+			})
+
+			By("Deleting the statefulset", func() {
```

Comment:
Let's move this By to AfterEach. It already used in all cases

Reply:
Actually it isnt used in the case:
> "should fail on creating the statefulset with invalid owner reference"